src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -293,25 +309,40 @@ export class ColorSwitchCC extends CommandClass {
 			direction: "none",
 		});
 
-		// TODO: Only do this if complete?  Where would we get
-		//	the previously discovered colors?
-		log.controller.logNode(node.id, {
-			endpoint: this.endpointIndex,
-			message: "querying Color Switch CC supported colors...",
-			direction: "outbound",
-		});
+		let supportedColors: SupportedColorTable;
+		if (complete) {
+			//	the previously discovered colors?
+			log.controller.logNode(node.id, {
+				endpoint: this.endpointIndex,
+				message: "querying Color Switch CC supported colors...",
+				direction: "outbound",
+			});
 
-		const colorsResponse = await api.getSupported();
+			const colorsResponse = await api.getSupported();
+			supportedColors = colorsResponse;
+		} else {
+			// TODO: Should we be storing supportedColor as a SupportedColorTable
+			//	instead of spreading it out on propertyKey?
+			supportedColors = { ...DefaultSupportedColorTable };
+			const valueDB = this.getValueDB();
+			for (const key of SupportedColorKeys) {
+				const valueId: ValueID = getSupportedColorValueID(
+					this.endpointIndex,
+					key,
+				);
+				const isSupported = valueDB.getValue<boolean>(valueId);
+				supportedColors[key] = isSupported ?? false;
+			}
+		}
```

Comment:
Yeah, store it as an object or Map or whatever makes the code the easiest. The Value ID should be marked as internal:
Add a constructor to `ColorSwitchCC`:
```ts
	public constructor(driver: Driver, options: CommandClassOptions) {
		super(driver, options);
		this.registerValue(getSupportedColorValueID(this.endpointIndex).property, true);
	}
```

Reply:
I think now I want to keep them as-is.  It is useful to expose what colors are supported, so the user knows not to try setting colors that will have no effect.